hilti/toolchain/include/ast/attribute.h

```diff
@@ -72,19 +73,40 @@ class Attribute : public NodeBase {
     template<typename T>
     Result<T> valueAs() const {
         if constexpr ( std::is_same<T, Expression>::value ) {
+            if ( ! hasValue() )
+                return result::Error(hilti::util::fmt("attribute '%s' requires an expression", _tag));
+
             if ( auto e = value().tryAs<Expression>() )
                 return *e;
 
             return result::Error(hilti::util::fmt("value for attribute '%s' must be an expression", _tag));
         }
 
         if constexpr ( std::is_same<T, std::string>::value ) {
+            if ( ! hasValue() )
+                return result::Error(hilti::util::fmt("attribute '%s' requires a string", _tag));
+
             if ( auto e = value().tryAs<expression::Ctor>() )
                 if ( auto s = e->ctor().tryAs<ctor::String>() )
                     return s->value();
             return result::Error(hilti::util::fmt("value for attribute '%s' must be a string", _tag));
         }
 
+        if constexpr ( std::is_same<T, int64_t>::value ) {
```

Comment:
We could use `std::is_integral_v<T>` here.

Originally I was about to suggest this so this function also works with `integer::safe<T>`, but it looks like it currently doesn't implement that trait. Might be worth being more general here nevertheless.

Reply:
Not sure, it's meant to get an `int64_t` ... :)